packages/ra-ui-materialui/src/field/BooleanField.tsx

```diff
@@ -37,19 +37,17 @@ interface Props extends FieldProps {
     valueLabelFalse?: string;
 }
 
-interface EnhancedProps extends WithStyles<typeof styles> {}
-
 export const BooleanField: SFC<
-    Props & InjectedFieldProps & EnhancedProps & TypographyProps
+    Props & InjectedFieldProps & TypographyProps
 > = ({
     className,
-    classes,
     source,
     record = {},
     valueLabelTrue,
     valueLabelFalse,
     ...rest
 }) => {
+    const classes = useStyles({});
```

Comment:
Users should be able to override classes. Pass `classesOverride` to the useStyles like this:

```suggestion
    const classes = useStyles({ classes: classesOverride });
```